--- a/src/jumbo.c
+++ b/src/jumbo.c
@@ -256,7 +256,7 @@
 #endif
 
 #if NEED_STRCASECMP_NATIVE
-int strcasecmp(char *dst, char *src) {
+int strcasecmp(const char *dst, const char *src) {
 	int f,l;
 	do {
 		if ( ((f = (unsigned char)(*(dst++))) >= 'A') && (f <= 'Z') )
@@ -269,7 +269,7 @@
 #endif
 
 #if NEED_STRNCASECMP_NATIVE
-int strncasecmp(char *dst, char *src, size_t count) {
+int strncasecmp(const char *dst, const char *src, size_t count) {
 	if (count) {
 		int f,l;
 		do {
